chips/nrf52/src/ppi.rs

```diff
@@ -0,0 +1,170 @@
+    pub ch14_eep: ReadWrite<u32, EventEndPoint::Register>, //0x580
+    pub ch14_tep: ReadWrite<u32, TaskEndPoint::Register>, //0x584
+    pub ch15_eep: ReadWrite<u32, EventEndPoint::Register>, //0x588
+    pub ch15_tep: ReadWrite<u32, TaskEndPoint::Register>, //0x58C
+    pub ch16_eep: ReadWrite<u32, EventEndPoint::Register>, //0x590
+    pub ch16_tep: ReadWrite<u32, TaskEndPoint::Register>, //0x594
+    pub ch17_eep: ReadWrite<u32, EventEndPoint::Register>, //0x598
+    pub ch17_tep: ReadWrite<u32, TaskEndPoint::Register>, //0x59C
+    pub ch18_eep: ReadWrite<u32, EventEndPoint::Register>, //0x5A0
+    pub ch18_tep: ReadWrite<u32, TaskEndPoint::Register>, //0x5A4
+    pub ch19_eep: ReadWrite<u32, EventEndPoint::Register>, //0x5A8
+    pub ch19_tep: ReadWrite<u32, TaskEndPoint::Register>, //0x5AC
+    _reserved2: [u32; 148],                               //0x5AC - 0x800
+    pub chg: [ReadWrite<u32, Channel::Register>; 6],      //0x800 - 0x814
+    _reserved3: [u32; 62],                                //0x814 - 0x910
+    pub fork_tep: [ReadWrite<u32, TaskEndPoint::Register>; 32], //0x910 - 0x98C
+}
+
+register_bitfields! [u32,
+    Control [
+        ENABLE OFFSET(0) NUMBITS(1)
+    ],
+    Channel [
+         CH0 OFFSET(0) NUMBITS(1),
+         CH1 OFFSET(1) NUMBITS(1),
+         CH2 OFFSET(2) NUMBITS(1),
+         CH3 OFFSET(3) NUMBITS(1),
+         CH4 OFFSET(4) NUMBITS(1),
+         CH5 OFFSET(5) NUMBITS(1),
+         CH6 OFFSET(6) NUMBITS(1),
+         CH7 OFFSET(7) NUMBITS(1),
+         CH8 OFFSET(8) NUMBITS(1),
+         CH9 OFFSET(9) NUMBITS(1),
+         CH10 OFFSET(10) NUMBITS(1),
+         CH11 OFFSET(11) NUMBITS(1),
+         CH12 OFFSET(12) NUMBITS(1),
+         CH13 OFFSET(13) NUMBITS(1),
+         CH14 OFFSET(14) NUMBITS(1),
+         CH15 OFFSET(15) NUMBITS(1),
+         CH16 OFFSET(16) NUMBITS(1),
+         CH17 OFFSET(17) NUMBITS(1),
+         CH18 OFFSET(18) NUMBITS(1),
+         CH19 OFFSET(19) NUMBITS(1),
+         CH20 OFFSET(20) NUMBITS(1),
+         CH21 OFFSET(21) NUMBITS(1),
+         CH22 OFFSET(22) NUMBITS(1),
+         CH23 OFFSET(23) NUMBITS(1),
+         CH24 OFFSET(24) NUMBITS(1),
+         CH25 OFFSET(25) NUMBITS(1),
+         CH26 OFFSET(26) NUMBITS(1),
+         CH27 OFFSET(27) NUMBITS(1),
+         CH28 OFFSET(28) NUMBITS(1),
+         CH29 OFFSET(29) NUMBITS(1),
+         CH30 OFFSET(30) NUMBITS(1),
+         CH31 OFFSET(31) NUMBITS(1)
+    ],
+    TaskEndPoint [
+        ENABLE OFFSET(0) NUMBITS(1)
+    ],
+    EventEndPoint [
+        ENABLE OFFSET(0) NUMBITS(1)
+    ]
+];
+
+pub struct PPIStruct {
+    regs: *const PPIRegs,
+}
+
+pub static mut PPI: PPIStruct = PPIStruct::new();
+
+impl PPIStruct {
+    pub const fn new() -> PPIStruct {
+        PPIStruct {
+            regs: PPI_BASE as *const PPIRegs,
+        }
+    }
+
+    pub fn enable(&self, channels: FieldValue<u32, Channel::Register>) {
+        let regs = unsafe { &*self.regs };
+        regs.chenset.write(channels);
```

Comment:
Curious as to why `write` and not `modify` as it would overwrite every other value in the register as well?

Reply:
Writing `0` to `CHENSET` has no effect. It will only `SET` the values which are `1`.